optimum/exporters/onnx/__main__.py

```diff
@@ -74,20 +75,17 @@ def main():
         task, args.model, framework=args.framework, cache_dir=args.cache_dir, trust_remote_code=args.trust_remote_code
     )
 
-    if task + "-with-past" in TasksManager.get_supported_tasks_for_model_type(
+    if task != "stable-diffusion" and task + "-with-past" in TasksManager.get_supported_tasks_for_model_type(
         model.config.model_type.replace("_", "-"), "onnx"
     ):
-        if args.task == "auto":  # Make -with-past the default if --task was not explicitely specified
+        if task == "auto":  # Make -with-past the default if --task was not explicitely specified
```

Comment:
It will never happen no?
I think what you want here is `args.task`.

Reply:
Good catch !